pkg/util/must/must.go

```diff
@@ -0,0 +1,450 @@
+// Copyright 2023 The Cockroach Authors.
+//
+// Use of this software is governed by the Business Source License
+// included in the file licenses/BSL.txt.
+//
+// As of the Change Date specified in that file, in accordance with
+// the Business Source License, use of this software will be governed
+// by the Apache License, Version 2.0, included in the file
+// licenses/APL.txt.
+
+// Package must provides a convenient API for runtime assertions.
+//
+// Runtime assertions are great: they build confidence in assumptions and
+// invariants, plug gaps in test coverage, and run in every single CRDB instance
+// across all environments and workloads. They can detect bugs in exceedingly
+// rare corner cases, and help with debugging incidents.
+//
+// We should liberally sprinkle assertions everywhere. This package makes that
+// easy. It provides familiar Testify-style assertions, similar to those used in
+// tests, which are cheap enough to not impact performance in the common case.
+//
+// In non-release builds (including roachprod and roachtest clusters), assertion
+// failures kill the process (with a stack trace) to ensure they are noticed. In
+// Go tests, they instead panic to allow other tests to run.
```

Comment:
That's almost the same, if one test panics it kills the entire bazel shard. There is too little per-test isolation in Go; especially if the failure is off the main goroutine.

Reply:
Actually no, not if we panic off the main goroutine -- let's just `log.Fatalf`.